apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java

```diff
@@ -295,6 +292,25 @@
             "Consider increasing the threshold or disabling stack trace collection altogether.")
         .buildWithDefault(Collections.<MethodMatcher>emptyList());
 
+    private final ConfigurationOption<TimeDuration> traceMethodsDurationThreshold = TimeDurationValueConverter.durationOption("ms")
+        .key("trace_methods_duration_threshold")
+        .tags("added[1.6.0]")
+        .configurationCategory(CORE_CATEGORY)
+        .description("If <<config-trace-methods, `trace_methods`>> config option is set, provides a threshold to limit spans based on \n" +
+            "duration. When set to a value greater than 0, spans representing methods traced based on `trace_methods` will be discarded " +
+            "by default.\n" +
+            "Such methods will be traced and reported if one of the following applies:\n" +
+            " - This method's duration crossed the configured threshold.\n" +
+            " - This method ended with Exception.\n" +
+            " - A method executed as part of the execution of this method crossed the threshold or ended with Exception.\n" +
+            " - A \"forcibly-traced method\" (e.g. DB queries, HTTP exits, custom) was executed during the execution of this method.\n" +
+            "Set to 0 to disable.\n" +
+            "\n" +
+            "NOTE: Transaction are never discarded, regardless of their duration. This configuration affects only spans.\n" +
+            "In order not to break span references, all spans leading to an async operations are never discarded, regardless \n" +
+            "of their duration.\n")
+        .buildWithDefault(TimeDuration.of("0ms"));
```

Comment:
Should we enhance TimeDuration to add support for `us` (µs)?

Reply:
Not sure if sub-millisecond is enhancing enough. I imagine someone using `trace_methods` with wildcards and hoping to get performance costs down shouldn't configure microseconds thresholds.